libraries/extractor/src/main/java/androidx/media3/extractor/ts/TsExtractor.java

```diff
@@ -323,7 +323,14 @@ public void release() {
     }
 
     if (!fillBufferWithAtLeastOnePacket(input)) {
-      return RESULT_END_OF_INPUT;
+      if (mode != MODE_HLS) {
```

Comment:
I've seen this PR, as it is, would break HLS mode, because of the presence of the next segment. However I think the logic of the PR should still apply to the very last segment of a VOD HLS, which currently misses the last frame.

Let me investigate a bit further on this.

Reply:
Of course when I say "would break" I mean without the if statement above

Reply:
I've added the HLS case as well.